client/bastionai/learner.py

```diff
@@ -18,22 +17,26 @@ def __init__(
         train_dataloader: DataLoader,
         test_dataloader: DataLoader,
         description: Optional[str] = None,
+        dataset_name: Optional[str] = None,
         secret: Optional[bytes] = None,
     ) -> None:
         if train_dataloader.batch_size != test_dataloader.batch_size:
-            raise Exception("Train and test dataloaders must use the same batch size.")
+            raise Exception(
+                "Train and test dataloaders must use the same batch size.")
         self.train_dataset_ref = client.send_dataset(
             train_dataloader.dataset,
             description
             if description is not None
             else type(train_dataloader.dataset).__name__,
+            dataset_name,
```

Comment:
It sounds weird to have an optional name and a automatically generated description at the same time
I think it would be best to automatically generate the name when not provided and have an optional description that remains blank (it could also be set equal to the name) when not provided

Reply:
What do you mean automatically generated description?
You mean they do not have to provide any description?

Reply:
if description is not None else type(train_dataloader.dataset).__name__
            
            In words, if description is not provided fill it with the name of the class

Reply:
Ohhh, so you'd rather us use the description as the name in the telemetry?